src/config.c

```diff
@@ -854,41 +872,19 @@ void configSetCommand(client *c) {
         /* Sanity check of single arguments, so that we either refuse the
          * whole configuration string or accept it all, even if a single
          * error in a single client class is present. */
-        for (j = 0; j < vlen; j++) {
-            if ((j % 4) == 0) {
-                int class = getClientTypeByName(v[j]);
-                if (class == -1 || class == CLIENT_TYPE_MASTER)
-                    break;
-            } else if ((j % 4) == 3) {
-                char *endptr;
-                long l = strtoll(v[j],&endptr,10);
-                if (l < 0 || *endptr != '\0')
-                    break;
-            } else {
-                memtoull(v[j], &err);
-                if (err)
-                    break;
+        for (j = 0; j < vlen; j += 4) {
+            if (updateClientOutputBufferLimit(&v[j], &errstr, 0) == C_ERR) {
+                sdsfreesplitres(v, vlen);
+                goto badfmt;
             }
         }
-        if (j < vlen) {
-            sdsfreesplitres(v,vlen);
-            goto badfmt;
-        }
-        
         /* Finally set the new config */
         for (j = 0; j < vlen; j += 4) {
-            int class;
-            unsigned long long hard, soft;
-            int soft_seconds;
-
-            class = getClientTypeByName(v[j]);
-            hard = memtoull(v[j+1],NULL);
-            soft = memtoull(v[j+2],NULL);
-            soft_seconds = strtoll(v[j+3],NULL,10);
-
-            server.client_obuf_limits[class].hard_limit_bytes = hard;
-            server.client_obuf_limits[class].soft_limit_bytes = soft;
-            server.client_obuf_limits[class].soft_limit_seconds = soft_seconds;
+            if (updateClientOutputBufferLimit(&v[j], &errstr, 1) == C_ERR) {
```

Comment:
Not quite sure here

Reply:
What aren't you sure about?

Reply:
Not sure if it needs to be changed here. 
`== C_ERR` that part never reached.

Reply:
I see. Yes no need to check for error.
A better approach might be to change the `updateClientOutputBufferLimit()` to always apply (remove the apply argument) and instead pass the output variable (`server.client_obuf_limits`). In this case pass some temp `clientBufferLimitsConfig` array. And if there's no error in the call above then copy the result to `server.client_obuf_limits`. This way you don't need to call the function twice, and don't need to check for the result a second time.